### PR TITLE
Replaced support libraries with AndroidX and made some necessary chan…

### PR DESCRIPTION
**Pull request by moksh-mahajan**
…ges for the upgrade

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MifosBaseActivity.java [hunk 1]
```diff
@@ -122,9 +122,12 @@ public void hideProgress() {
 
     public void hideKeyboard(View view) {
         InputMethodManager inputManager = (InputMethodManager) this
-                 .getSystemService(Context.INPUT_METHOD_SERVICE);
-        inputManager.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), InputMethodManager
-                 .RESULT_UNCHANGED_SHOWN);
+                .getSystemService(Context.INPUT_METHOD_SERVICE);
+        View focusedView = this.getCurrentFocus();
+        if (focusedView != null) {
```
Comment (tarun0) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MifosBaseActivity.java]: 1. This change is not related to AndroidX porting. So, it shouldn't have been here. 
2. What if the view is null?

Comment (moksh-mahajan) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/core/MifosBaseActivity.java]: Actually it's a UI bug. I will create a separate issue for that.
